runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -241,15 +241,15 @@ local function on_line_impl(self, buf, line, is_spell_nav)
     end
 
     while line >= state.next_row do
-      local capture, node, metadata = state.iter()
+      local capture, node, metadata = state.iter(line)
+      metadata = metadata or {}
 
-      if capture == nil then
-        break
+      local range = { root_end_row + 1, 0, root_end_row + 1, 0 }
+      if node then
+        range = vim.treesitter.get_range(node, buf, metadata[capture])
       end
-
-      local range = vim.treesitter.get_range(node, buf, metadata[capture])
       local start_row, start_col, end_row, end_col = Range.unpack4(range)
-      local hl = highlighter_query.hl_cache[capture]
+      local hl = capture and highlighter_query.hl_cache[capture]
 
       local capture_name = highlighter_query:query().captures[capture]
```

Comment:
Index with `nil` error.

I think this whole block needs to be surrounded with `if capture`.

Reply:
roger